### PR TITLE
MM-50252_Refresh UI of the marketplace to improve first impressions

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
UI refresh of the Marketplace modal to improve first impressions of onboarding users.

UX design: https://www.figma.com/file/nwO2jTKNE1glTrDj5gefwr/App-Marketplace?node-id=1918%3A80279&t=K3OJinE5mjKhK2gU-1

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50252

#### Screenshots
Screen Shot 2023-03-27 at 3 23 14 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by annelieseklein** [comment]
@julmondragon
Looks good!

Plugins like Calls or To-do that are functional and don't need further configuration in order to be used should have a button that says "Edit configuration" instead of Configure - which still sends the user to that page in the System Console cc: @esethna 

Screen Shot 2023-03-27 at 5 07 46 PM

https://www.figma.com/file/nwO2jTKNE1glTrDj5gefwr/App-Marketplace?node-id=2156%3A89892&t=xSwDFfxwIZ4gt4nW-1

**Comment by esethna** [comment]
@annelieseklein I think we need to provide more detailed functional notes in order to implement this accurately. Specifically, what are the conditions when admin users will see the "Configure" button vs when users will see "Edit configuration"

Is it, when a plugin is "enabled" in the configuration page that it should be "Edit configuration"? Or are there more cases?

**Comment by esethna** [comment]
Actually, @annelieseklein, can we file this as a separate ticket with the details ^ so we unblock this PR?

File: webapp/channels/src/components/widgets/inputs/input/input.tsx [hunk 1]
```diff
@@ -3,10 +3,14 @@
 
 import React, {useState, useEffect} from 'react';
 import {useIntl} from 'react-intl';
-
 import classNames from 'classnames';
 
-import {ItemStatus} from 'utils/constants';
```
Comment (julmondragon) [inline webapp/channels/src/components/widgets/inputs/input/input.tsx:9]: Enhancement to the Input component to provide the 'clearable' feature.

File: webapp/platform/components/src/generic_modal/generic_modal.tsx [hunk 2]
```diff
@@ -34,13 +34,17 @@ export type Props = {
     enforceFocus?: boolean;
```
Comment (julmondragon) [inline webapp/platform/components/src/generic_modal/generic_modal.tsx:34]: Enhancement to the GenericModal component to support the header and footer variations specified in UX Patterns design.

**Comment by annelieseklein** [comment]
Screen Shot 2023-03-28 at 11 05 23 AM
I believe in other cases in the product we dont' disable it in the loading state

File: webapp/channels/src/components/admin_console/license_settings/license_settings.test.tsx [hunk 3]
```diff
@@ -181,7 +181,7 @@ describe('components/admin_console/license_settings/LicenseSettings', () => {
         const instance = wrapper.instance();
 
         // First start trial
-        actions.requestTrialLicense = jest.fn().mockImplementation(() => Promise.resolve({percentage: 1, error: null}));
+        actions.requestTrialLicense = jest.fn().mockImplementation(() => ({percentage: 1, error: null}));
```
Comment (pvev) [inline webapp/channels/src/components/admin_console/license_settings/license_settings.test.tsx:184]: is this change intended?

File: webapp/channels/src/components/plugin_marketplace/marketplace_list/marketplace_list.tsx [hunk 4]
```diff
@@ -1,107 +1,116 @@
...
-                })
-                }
-                <NavigationRow
-                    page={this.state.page}
-                    total={this.props.listing.length}
-                    maximumPerPage={ITEMS_PER_PAGE}
-                    onNextPageButtonClick={this.nextPage}
-                    onPreviousPageButtonClick={this.previousPage}
-                />
+        return [...listing].
```
Comment (pvev) [inline webapp/channels/src/components/plugin_marketplace/marketplace_list/marketplace_list.tsx:47]: nit: splitting this logic a little bit as it was before makes it easier to read

**Review by pvev — APPROVED** [review]
LGTM! thanks @julmondragon :shipit:

**Comment by pvev** [comment]
notice that the zendesk plugin 
Screenshot 2023-03-28 at 22 27 07
image is not loading in spinwick

File: webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx [hunk 5]
```diff
@@ -1,283 +1,269 @@
...
-        const {error} = await this.props.actions.fetchListing();
-        this.setState({loading: false, serverError: error});
-    }
+        trackEvent('plugins', 'ui_marketplace_opened', {from: openedFrom});
 
-    close = (): void => {
-        trackEvent('plugins', 'ui_marketplace_closed');
-        this.props.actions.closeModal();
-    }
+        if (firstAdminVisitMarketplaceStatus) {
```
Comment (marianunez) [inline webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx:106]: Noticed that before this was `if (!this.props.firstAdminVisitMarketplaceStatus)` which seems it was a bug and you are fixing it here removing the not condition?

**Review by marianunez — APPROVED** [review]
Great refactoring with nice additions to existing generic components 🚀 

Non-blocking comments below

File: webapp/channels/src/components/admin_console/license_settings/license_settings.test.tsx (hunk 3 above)
Comment (julmondragon) [reply webapp/channels/src/components/admin_console/license_settings/license_settings.test.tsx]: Not really, probably part of the multiple conflicts I've been solving. Removed.

File: webapp/channels/src/components/generic_modal.scss [hunk 6]
```diff
@@ -133,7 +138,13 @@
                 }
 
                 .GenericModal__header {
+                    display: flex;
+                    flex: 1;
+                    flex-direction: row;
+                    align-items: center;
+                    justify-content: space-between;
                     padding: 0;
+                    margin-right: 32px;
```
[the remark this answers is not in the document]
Comment (julmondragon) [reply webapp/channels/src/components/generic_modal.scss:147]: Correct, this is applied when using the GenericModal with `compassDesign` enabled, validated them for compatibility.

File: webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx (hunk 5 above)
Comment (julmondragon) [reply webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx]: Nice catch! I was tricked by the config name. Updated along with the const name to make it more clear.

File: webapp/channels/src/components/widgets/inputs/input/input.tsx [hunk 7]
```diff
@@ -32,6 +36,9 @@ interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
     useLegend?: boolean;
     customMessage?: CustomMessageInputType;
     inputSize?: SIZE;
+    clearable?: boolean;
+    clearableTooltipText?: string;
+    onClear?: () => void;
```
[the remark this answers is not in the document]
Comment (marianunez) [reply webapp/channels/src/components/widgets/inputs/input/input.tsx:41]: Nice tests, thanks!

File: webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx (hunk 5 above)
Comment (marianunez) [reply webapp/channels/src/components/plugin_marketplace/marketplace_modal.tsx]: Good renaming - it was also confusing for me when I first saw it and thought the logic was wrong. 

We need to test this case of the tracking being emitted only once for admins. @julmondragon could you include this in the QA steps to be sure we cover it? cc/ @stevemudie

**Review by stevemudie — APPROVED** [review]
Looks great @julmondragon!
